e2store/src/e2store/types.rs

```diff
@@ -0,0 +1,156 @@
+use anyhow::{anyhow, ensure};
+use ssz_derive::{Decode, Encode};
+
+pub const HEADER_SIZE: u16 = 8;
+
+/// Represents an e2store `Entry`
+#[derive(Default, Debug, Eq, PartialEq, Clone)]
+pub struct Entry {
```

Comment:
> Somewhat related to this, does max size depends on the type of era file or is it always u32?

The max size of a e2store entry is u32, this doesn't include the length of the header

Reply:
I am open to making a change like this in the future, but I think I will leave it for now. I don't think we can remove the length check for `deserialize` as we could possibly be reading 3rd party era files which don't follow rules

Reply:
> I don't think we can remove the length check for `deserialize` as we could possibly be reading 3rd party era files which don't follow rules

It could be removed from deserialize because it would call constructor that would do it for us (meaning, we would still do it, just indirectly).